src/js/components/FileInput/FileInput.js

```diff
@@ -219,7 +222,17 @@ const FileInput = forwardRef(
               const fileList = event.target.files;
               const nextFiles = multiple ? [...files] : [];
               for (let i = 0; i < fileList.length; i += 1) {
-                nextFiles.push(fileList[i]);
+                // avoid duplicates
+                let dup = false;
+                for (let j = 0; j < files.length; j += 1) {
```

Comment:
any reason to do traditional for loop vs using `.filter()` method?

Reply:
where you only return the file to the results array if it's not a duplicate

Reply:
could be done that way. I was in `for` mode because the FileList object isn't a true Array, so you have to use `for` with it.

Reply:
switched to `filter`, less code